certchain/certmanager/helpers_test.go

```diff
@@ -93,15 +94,28 @@ func (p *stubProducer) Stop() { close(p.out) }
 var _ certmanager.Cache = (*stubCache)(nil)
 
 type stubCache struct {
-	OnWrite chan *certchain.AugmentedChain
+	avail    chan bool
+	chainMap map[string]*certchain.AugmentedChain
 }
 
 func newStubCache() *stubCache {
-	return &stubCache{make(chan *certchain.AugmentedChain, 1)}
+	return &stubCache{make(chan bool, 1), make(map[string]*certchain.AugmentedChain)}
+}
+
+func (c *stubCache) Read(digest string) (*certchain.AugmentedChain, error) {
+	<-c.avail
+	if c.chainMap[digest] == nil {
```

Comment:
Optional: This is good enough, but you can also write:

```go
if _, ok := c.chainMap[digest]; !ok {
  // ...
}
```

and this code might express the intention more clearly.

Reply:
Done.